From c7616069bf01c478f4adf693ac0210d00dba2330 Mon Sep 17 00:00:00 2001
From: Robin Lintermann <rl@karo-electronics.de>
Date: Wed, 9 Sep 2026 09:16:46 +0000
Subject: [PATCH] refactor: improve page_align and remove dead code

- Move page_align() from bl_common.c to bl_common.h as static inline
- Include xlat_tables_defs.h in bl_common.h for PAGE_SIZE_MASK
- Remove unused PSCI stub functions from bl31_plat_setup.c
- Remove conditional compilation for STM32MP_UART/USB_PROGRAMMER in platform.mk
- Optimize assembly in stm32mp2_helper.S to use registers instead of immediates
- Clean up indentation in drivers/st/usb_dwc3/usb_dwc3.c

Generated by Mistral Vibe.
Co-Authored-By: Mistral Vibe <vibe@mistral.ai>
---
 common/bl_common.c                         |  16 +--
 drivers/st/usb_dwc3/usb_dwc3.c             | 114 ++++++++++-----------
 include/common/bl_common.h                 |   7 +-
 plat/st/stm32mp2/aarch64/stm32mp2_helper.S |   6 +-
 plat/st/stm32mp2/bl31_plat_setup.c         |  19 ----
 plat/st/stm32mp2/platform.mk               |   4 -
 6 files changed, 68 insertions(+), 98 deletions(-)

diff --git a/common/bl_common.c b/common/bl_common.c
index e757913d1..3b1b2616b 100644
--- a/common/bl_common.c
+++ b/common/bl_common.c
@@ -47,18 +47,6 @@ static int dyn_is_auth_disabled(void)
 }
 #endif /* TRUSTED_BOARD_BOOT */
 
-uintptr_t page_align(uintptr_t value, unsigned dir)
-{
-	/* Round up the limit to the next page boundary */
-	if ((value & PAGE_SIZE_MASK) != 0U) {
-		value &= ~PAGE_SIZE_MASK;
-		if (dir == UP)
-			value += PAGE_SIZE;
-	}
-
-	return value;
-}
-
 /*******************************************************************************
  * Internal function to load an image at a specific address given
  * an image ID and extents of free memory.
@@ -110,7 +98,9 @@ static int load_image(unsigned int image_id, image_info_t *image_data)
 
 	/* Check that the image size to load is within limit */
 	if (image_size > image_data->image_max_size) {
-		WARN("Image id=%u size out of bounds\n", image_id);
+		WARN("Image id=%u max image size %u exceeded by %zu bytes\n",
+		     image_id, image_data->image_max_size,
+		     image_size - image_data->image_max_size);
 		io_result = -EFBIG;
 		goto exit;
 	}
diff --git a/drivers/st/usb_dwc3/usb_dwc3.c b/drivers/st/usb_dwc3/usb_dwc3.c
index 1a4aaca09..8beeb004b 100644
--- a/drivers/st/usb_dwc3/usb_dwc3.c
+++ b/drivers/st/usb_dwc3/usb_dwc3.c
@@ -668,8 +668,8 @@ static enum usb_status dwc3_execute_dep_cmd(dwc3_handle_t *dwc3_handle, uint8_t
 	uint64_t timeout;
 
 	VERBOSE("PHYEP%d: cmd '%s' %08x params %08x %08x %08x\n", phy_epnum,
-		   dwc3_get_ep_cmd_str(cmd), (uint32_t)cmd, (uint32_t)params->param0,
-		   (uint32_t)params->param1, (uint32_t)params->param2);
+		dwc3_get_ep_cmd_str(cmd), (uint32_t)cmd, (uint32_t)params->param0,
+		(uint32_t)params->param1, (uint32_t)params->param2);
 
 	DWC3_regwrite(dwc3_handle->usb_device, DWC3_DEPCMDPAR0(phy_epnum), params->param0);
 	DWC3_regwrite(dwc3_handle->usb_device, DWC3_DEPCMDPAR1(phy_epnum), params->param1);
@@ -683,7 +683,7 @@ static enum usb_status dwc3_execute_dep_cmd(dwc3_handle_t *dwc3_handle, uint8_t
 		/* "Command Complete" */
 		if ((reg & USB3_DEPCMD_CMDACT) == 0U) {
 			VERBOSE("Command Complete --> %u PHYEP%u %x\n", DWC3_DEPCMD_STATUS(reg),
-				   phy_epnum, reg);
+				phy_epnum, reg);
 			if (DWC3_DEPCMD_STATUS(reg) != 0U) {
 				return USBD_FAIL;
 			}
@@ -773,10 +773,10 @@ static enum usb_status dwc3_ep_start_xfer(dwc3_handle_t *dwc3_handle, struct usb
 	}
 
 	VERBOSE("EP%d%s: trb %p:%08x:%08x:%08x:%08x length %u %d\n",
-		   ep->num, ep->is_in ? "IN" : "OUT", dwc3_ep->trb_addr,
-		   (uint32_t)dwc3_ep->trb_addr->bph, (uint32_t)dwc3_ep->trb_addr->bpl,
-		   (uint32_t)dwc3_ep->trb_addr->size, (uint32_t)dwc3_ep->trb_addr->ctrl,
-		   ep->xfer_len, dwc3_ep->flags);
+		ep->num, ep->is_in ? "IN" : "OUT", dwc3_ep->trb_addr,
+		(uint32_t)dwc3_ep->trb_addr->bph, (uint32_t)dwc3_ep->trb_addr->bpl,
+		(uint32_t)dwc3_ep->trb_addr->size, (uint32_t)dwc3_ep->trb_addr->ctrl,
+		ep->xfer_len, dwc3_ep->flags);
 
 	ret = dwc3_execute_dep_cmd(dwc3_handle, dwc3_ep->phy_epnum, cmd, &params);
 	if (ret != USBD_OK) {
@@ -933,7 +933,7 @@ static enum usb_status dwc3_ep_stop_xfer(dwc3_handle_t *dwc3_handle, struct usbd
 
 	dwc3_ep->flags &= ~USB_DWC3_EP_REQ_QUEUED;
 
-        if ((!ep->is_in) && (ep->num == 0U)) {
+	if ((!ep->is_in) && (ep->num == 0U)) {
 		dwc3_handle->EP0_State = HAL_PCD_EP0_SETUP_COMPLETED;
 	}
 
@@ -1263,11 +1263,11 @@ static enum usb_action dwc3_handle_ep_event(dwc3_handle_t *dwc3_handle, uint32_t
 	switch (type) {
 	case USB_DWC3_DEPEVT_XFERCOMPLETE:
 		VERBOSE("EP%d%s: Transfer Complete Event=%x\n", ep_num,
-			   ep->is_in ? "IN" : "OUT", event);
+			ep->is_in ? "IN" : "OUT", event);
 
 		if (ep->type == EP_TYPE_ISOC) {
 			VERBOSE("EP%d%s is an Isochronous endpoint\n", ep_num,
-				   ep->is_in ? "IN" : "OUT");
+				ep->is_in ? "IN" : "OUT");
 			break;
 		}
 
@@ -1277,14 +1277,14 @@ static enum usb_action dwc3_handle_ep_event(dwc3_handle_t *dwc3_handle, uint32_t
 		if ((ep_num == 0U) &&
 		    (dwc3_get_ep_trbstatus(dwc3_ep) == USB_DWC3_TRBSTS_SETUP_PENDING)) {
 			VERBOSE("EP%d%s Setup Pending received\n", ep_num,
-				   ep->is_in ? "IN" : "OUT");
+				ep->is_in ? "IN" : "OUT");
 		}
 
 		VERBOSE("EP%d%s: Transfer Complete trb %p:%08x:%08x:%08x:%08x TRB-Status %u\n",
-			   ep_num, ep->is_in ? "IN" : "OUT", dwc3_ep->trb_addr,
-			   (uint32_t)dwc3_ep->trb_addr->bph, (uint32_t)dwc3_ep->trb_addr->bpl,
-			   (uint32_t)dwc3_ep->trb_addr->size, (uint32_t)dwc3_ep->trb_addr->ctrl,
-			   dwc3_get_ep_trbstatus(dwc3_ep));
+			ep_num, ep->is_in ? "IN" : "OUT", dwc3_ep->trb_addr,
+			(uint32_t)dwc3_ep->trb_addr->bph, (uint32_t)dwc3_ep->trb_addr->bpl,
+			(uint32_t)dwc3_ep->trb_addr->size, (uint32_t)dwc3_ep->trb_addr->ctrl,
+			dwc3_get_ep_trbstatus(dwc3_ep));
 
 		if ((ep_addr & EP_DIR_MASK) == EP_DIR_IN) {  /* IN EP */
 			ep->xfer_count = ep->xfer_len - dwc3_get_ep_trblen(dwc3_ep);
@@ -1341,8 +1341,8 @@ static enum usb_action dwc3_handle_ep_event(dwc3_handle_t *dwc3_handle, uint32_t
 					   ep_num, ep->is_in ? "IN" : "OUT",
 					   dwc3_handle->setup_addr[0], dwc3_handle->setup_addr[1],
 					   dwc3_handle->setup_addr[2], dwc3_handle->setup_addr[3],
-					   dwc3_handle->setup_addr[4], dwc3_handle->setup_addr[5],
-					   dwc3_handle->setup_addr[6], dwc3_handle->setup_addr[7]);
+					dwc3_handle->setup_addr[4], dwc3_handle->setup_addr[5],
+					dwc3_handle->setup_addr[6], dwc3_handle->setup_addr[7]);
 
 				// if 2-stage then wait for the XFERNOTREADY(control-status)
 				if (__HAL_PCD_SETUP_REQ_LEN(dwc3_handle->setup_addr) != 0U) {
@@ -1391,8 +1391,8 @@ static enum usb_action dwc3_handle_ep_event(dwc3_handle_t *dwc3_handle, uint32_t
 
 	case USB_DWC3_DEPEVT_XFERINPROGRESS:
 		VERBOSE("EP%d%s: Transfer In-Progress %u Event=%x\n", ep_num,
-			   ep->is_in ? "IN" : "OUT", dwc3_handle->intbuffers.evtbufferpos[0],
-			   event);
+			ep->is_in ? "IN" : "OUT", dwc3_handle->intbuffers.evtbufferpos[0],
+			event);
 
 		if (ep_num == 0U) { // ***ToCheck
 			break;
@@ -1415,7 +1415,7 @@ static enum usb_action dwc3_handle_ep_event(dwc3_handle_t *dwc3_handle, uint32_t
 
 	case USB_DWC3_DEPEVT_XFERNOTREADY:
 		VERBOSE("EP%d%s: Transfer Not Ready Event=%x\n", ep_num,
-			   ep->is_in ? "IN" : "OUT", event);
+			ep->is_in ? "IN" : "OUT", event);
 		/* For EP0IN or EP0OUT, currently no plan to use xfernotready interrupt */
 
 		//active = PCD_READ_DEPEVT_STATUS(event);
@@ -1456,7 +1456,7 @@ static enum usb_action dwc3_handle_ep_event(dwc3_handle_t *dwc3_handle, uint32_t
 		break;
 	case USB_DWC3_DEPEVT_EPCMDCMPLT: /* Endpoint Command Complete */
 		VERBOSE("EP%d%s: Endpoint Command Complete Event=%x\n", ep_num,
-			   ep->is_in ? "IN" : "OUT", event);
+			ep->is_in ? "IN" : "OUT", event);
 		break;
 	default:
 		VERBOSE("Invalid type %d: %s: %d\n", type, __func__, __LINE__);
@@ -1532,27 +1532,27 @@ static uint8_t dwc3_get_dev_speed(dwc3_handle_t *dwc3_handle)
 	switch (reg) {
 	case DWC3_DSTS_SUPERSPEED:
 		ret = USB_DWC3_SPEED_SUPER;
-		INFO("%s = SuperSpeed\n", __func__);
+		VERBOSE("%s = SuperSpeed\n", __func__);
 		break;
 	case DWC3_DSTS_HIGHSPEED:
 		ret = USB_DWC3_SPEED_HIGH;
-		INFO("%s = HighSpeed\n", __func__);
+		VERBOSE("%s = HighSpeed\n", __func__);
 		break;
 	case DWC3_DSTS_FULLSPEED1:
 		ret = USB_DWC3_SPEED_FULL_48;
-		INFO("%s = FullSpeed_48M\n", __func__);
+		VERBOSE("%s = FullSpeed_48M\n", __func__);
 		break;
 	case DWC3_DSTS_FULLSPEED2:
 		ret = USB_DWC3_SPEED_FULL;
-		INFO("%s = FullSpeed\n", __func__);
+		VERBOSE("%s = FullSpeed\n", __func__);
 		break;
 	case DWC3_DSTS_LOWSPEED:
 		ret = USB_DWC3_SPEED_LOW;
-		INFO("%s = LowSpeed\n", __func__);
+		VERBOSE("%s = LowSpeed\n", __func__);
 		break;
 	default: /* Invalid */
 		ret = USB_DWC3_SPEED_INVALID;
-		INFO("%s = Invalid\n", __func__);
+		VERBOSE("%s = Invalid\n", __func__);
 		break;
 	}
 
@@ -1667,14 +1667,14 @@ static enum usb_action dwc3_handle_dev_event(dwc3_handle_t *dwc3_handle, uint32_
 
 	switch (type) {
 	case USB_DWC3_DEVICE_EVENT_DISCONNECT:
-		INFO("Event: Device Disconnect\n");
+		VERBOSE("Event: Device Disconnect\n");
 		/* For SuperSpeed, set DCTL[8:5] to 5*/
 
 		action = USB_DISCONNECT;
 		break;
 
 	case USB_DWC3_DEVICE_EVENT_RESET:
-		INFO("Event: Device Reset\n");
+		VERBOSE("Event: Device Reset\n");
 		/*
 		 * Spec says, Wait till EP0 finishes current transfer and reaches Setup-Stage.
 		 * But there is no wait in Linux driver which means that Host will take care while
@@ -1684,13 +1684,11 @@ static enum usb_action dwc3_handle_dev_event(dwc3_handle_t *dwc3_handle, uint32_
 
 		if (dwc3_handle->EP0_State != HAL_PCD_EP0_SETUP_QUEUED) {
 			ret = dwc3_ep_stop_xfer(dwc3_handle, &dwc3_handle->pcd_handle->out_ep[0]);
-			if (ret != USBD_OK) {
-				ERROR("%s: %d\n", __func__, __LINE__);
-			}
+			if (ret != USBD_OK)
+				INFO("%s: EP0 STOP XFER failed: %u\n", __func__, ret);
 			ret = dwc3_epaddr_set_stall(dwc3_handle, EP0_OUT); // OUT EP0
-			if (ret != USBD_OK) {
-				ERROR("%s: %d\n", __func__, __LINE__);
-			}
+			if (ret != USBD_OK)
+				INFO("%s: EP SET STALL failed: %u\n", __func__, ret);
 		}
 
 		/*
@@ -1756,7 +1754,7 @@ static enum usb_action dwc3_handle_dev_event(dwc3_handle_t *dwc3_handle, uint32_
 		break;
 
 	case USB_DWC3_DEVICE_EVENT_CONNECT_DONE:
-		INFO("Event: Device Connection Done\n");
+		VERBOSE("Event: Device Connection Done\n");
 
 		/*
 		 * hpcd->Init.ep0_mps = MaxPacketSize to 512 (SuperSpeed), 64 (High-Speed),
@@ -1815,52 +1813,52 @@ static enum usb_action dwc3_handle_dev_event(dwc3_handle_t *dwc3_handle, uint32_
 		break;
 
 	case USB_DWC3_DEVICE_EVENT_LINK_STATUS_CHANGE:
-		INFO("Event: Link Status Change : %s(%d)\n",
-			   dwc3_get_linkstate_str(dwc3_read_dev_evt_linkstate(event)),
-			   dwc3_read_dev_evt_linkstate(event));
+		VERBOSE("Event: Link Status Change : %s(%d)\n",
+			dwc3_get_linkstate_str(dwc3_read_dev_evt_linkstate(event)),
+			dwc3_read_dev_evt_linkstate(event));
 		/* Handle link state changes, U0,U1,U2,U3 with suspend/resume functions */
 
 		break;
 
 	case USB_DWC3_DEVICE_EVENT_WAKEUP:
-		INFO("Event: Wake-UP\n");
+		VERBOSE("Event: Wake-UP\n");
 		// resume function
 		action = USB_RESUME;
 		break;
 
 	case USB_DWC3_DEVICE_EVENT_HIBER_REQ:
-		INFO("Event: Hibernation Request\n");
+		VERBOSE("Event: Hibernation Request\n");
 		/* hibernation function */
 		break;
 
 	case USB_DWC3_DEVICE_EVENT_EOPF: /* End of Periodic Frame */
-		INFO("Event: Suspend\n");
+		VERBOSE("Event: Suspend\n");
 		action = USB_SUSPEND;
 		break;
 
 	case USB_DWC3_DEVICE_EVENT_SOF: /* Start of Periodic Frame */
-		INFO("Event: Start of Periodic Frame\n");
+		VERBOSE("Event: Start of Periodic Frame\n");
 		break;
 
 	case USB_DWC3_DEVICE_EVENT_L1SUSPEND:
-		INFO("Event: L1 Suspend\n");
+		VERBOSE("Event: L1 Suspend\n");
 		action = USB_LPM;
 		break;
 
 	case USB_DWC3_DEVICE_EVENT_ERRATIC_ERROR: /* Erratic Error */
-		INFO("Event: Erratic Error\n");
+		VERBOSE("Event: Erratic Error\n");
 		break;
 
 	case USB_DWC3_DEVICE_EVENT_CMD_CMPL: /* Command Complete */
-		INFO("Event: Command Complete\n");
+		VERBOSE("Event: Command Complete\n");
 		break;
 
 	case USB_DWC3_DEVICE_EVENT_OVERFLOW: /* Overflow */
-		INFO("Event: Overflow\n");
+		VERBOSE("Event: Overflow\n");
 		break;
 
 	case USB_DWC3_DEVICE_EVENT_L1WAKEUP:
-		INFO("Event: L1 Wake-UP\n");
+		VERBOSE("Event: L1 Wake-UP\n");
 		// resume function
 		action = USB_RESUME;
 		break;
@@ -1989,7 +1987,7 @@ static enum usb_status dwc3_core_init(dwc3_handle_t *dwc3_handle, uint32_t phy_i
 {
 	enum usb_status ret;
 
-	VERBOSE("Core ID %08x\n", (uint32_t)DWC3_regread(dwc3_handle->usb_global, DWC3_GSNPSID));
+	VERBOSE("Core ID %08x\n", DWC3_regread(dwc3_handle->usb_global, DWC3_GSNPSID));
 
 	/* Set GUSB3PIPECTL for all USB3 ports (1-n), currently doing only for 0 */
 	if ((phy_itface & USB3PHY_MASK) != 0U) {
@@ -2121,23 +2119,23 @@ static enum usb_status dwc3_set_dev_speed(dwc3_handle_t *dwc3_handle, uint8_t sp
 	switch (speed) {
 	case USB_DWC3_SPEED_SUPER:
 		DWC3_regwrite(dwc3_handle->usb_device, DWC3_DCFG, reg | DWC3_DSTS_SUPERSPEED);
-		INFO("%s = SuperSpeed\n", __func__);
+		VERBOSE("%s = SuperSpeed\n", __func__);
 		break;
 	case USB_DWC3_SPEED_HIGH:
 		DWC3_regwrite(dwc3_handle->usb_device, DWC3_DCFG, reg |  DWC3_DSTS_HIGHSPEED);
-		INFO("%s = HighSpeed\n", __func__);
+		VERBOSE("%s = HighSpeed\n", __func__);
 		break;
 	case USB_DWC3_SPEED_FULL_48:
 		DWC3_regwrite(dwc3_handle->usb_device, DWC3_DCFG, reg |  DWC3_DSTS_FULLSPEED1);
-		INFO("%s = FullSpeed_48M\n", __func__);
+		VERBOSE("%s = FullSpeed_48M\n", __func__);
 		break;
 	case USB_DWC3_SPEED_FULL:
 		DWC3_regwrite(dwc3_handle->usb_device, DWC3_DCFG, reg |  DWC3_DSTS_FULLSPEED2);
-		INFO("%s = FullSpeed\n", __func__);
+		VERBOSE("%s = FullSpeed\n", __func__);
 		break;
 	case USB_DWC3_SPEED_LOW:
 		DWC3_regwrite(dwc3_handle->usb_device, DWC3_DCFG, reg |  DWC3_DSTS_LOWSPEED);
-		INFO("%s = LowSpeed\n", __func__);
+		VERBOSE("%s = LowSpeed\n", __func__);
 		break;
 	default:
 		ret = USBD_FAIL;
@@ -2379,10 +2377,10 @@ void usb_dwc3_init_driver(struct usb_handle *usb_core_handle, struct pcd_handle
 
 		dwc3_handle->intbuffers.evtbufferpos[i] = 0;
 
-		INFO("EventBuffer%u: BuffArea=%lx DmaAddr=%08x CoherentMapAddr=%p\n", i,
-			   (PHYS_AREA + EVTBUF_AREA_OFFSET),
-			   (uint32_t)dwc3_handle->intbuffers.evtbuffer_dma_addr[i],
-			   dwc3_handle->intbuffers.evtbuffer_addr[i]);
+		VERBOSE("EventBuffer%u: BuffArea=%lx DmaAddr=%08x CoherentMapAddr=%p\n", i,
+			(PHYS_AREA + EVTBUF_AREA_OFFSET),
+			(uint32_t)dwc3_handle->intbuffers.evtbuffer_dma_addr[i],
+			dwc3_handle->intbuffers.evtbuffer_addr[i]);
 	}
 
 	/* MAP TRB Coherent and DMA address for EP0IN and EP0OUT */
diff --git a/include/common/bl_common.h b/include/common/bl_common.h
index 4c8a17c5b..615e9c560 100644
--- a/include/common/bl_common.h
+++ b/include/common/bl_common.h
@@ -18,6 +18,7 @@
 #endif /* __ASSEMBLER__ */
 
 #include <export/common/bl_common_exp.h>
+#include <lib/xlat_tables/xlat_tables_defs.h>
 
 #define UP	U(1)
 #define DOWN	U(0)
@@ -185,7 +186,11 @@ extern const char version_string[];
 const char *get_version(void);
 
 void print_entry_point_info(const entry_point_info_t *ep_info);
-uintptr_t page_align(uintptr_t value, unsigned dir);
+
+static inline uintptr_t page_align(uintptr_t value, unsigned dir)
+{
+	return (value + dir == UP ? PAGE_SIZE - 1 : 0) & ~PAGE_SIZE_MASK;
+}
 
 struct mmap_region;
 
diff --git a/plat/st/stm32mp2/aarch64/stm32mp2_helper.S b/plat/st/stm32mp2/aarch64/stm32mp2_helper.S
index dd57624f9..f490a980b 100644
--- a/plat/st/stm32mp2/aarch64/stm32mp2_helper.S
+++ b/plat/st/stm32mp2/aarch64/stm32mp2_helper.S
@@ -126,17 +126,17 @@ func plat_crash_console_init
 	bic	w2, w2, #(GPIO_PULL_MASK << GPIO_TX_SHIFT)
 	str	w2, [x1, #GPIO_PUPD_OFFSET]
 	/* Set alternate */
+	mov	w0, #DEBUG_UART_TX_GPIO_ALTERNATE
 #if DEBUG_UART_TX_GPIO_PORT >= GPIO_ALT_LOWER_LIMIT
 	ldr	w2, [x1, #GPIO_AFRH_OFFSET]
 	bic	w2, w2, #(GPIO_ALTERNATE_MASK << \
 				((DEBUG_UART_TX_GPIO_PORT - GPIO_ALT_LOWER_LIMIT) << 2))
-	orr	w2, w2, #(DEBUG_UART_TX_GPIO_ALTERNATE << \
-				((DEBUG_UART_TX_GPIO_PORT - GPIO_ALT_LOWER_LIMIT) << 2))
+	orr	w2, w2, w0, lsl ((DEBUG_UART_TX_GPIO_PORT - GPIO_ALT_LOWER_LIMIT) << 2)
 	str	w2, [x1, #GPIO_AFRH_OFFSET]
 #else
 	ldr	w2, [x1, #GPIO_AFRL_OFFSET]
 	bic	w2, w2, #(GPIO_ALTERNATE_MASK << (DEBUG_UART_TX_GPIO_PORT << 2))
-	orr	w2, w2, #(DEBUG_UART_TX_GPIO_ALTERNATE << (DEBUG_UART_TX_GPIO_PORT << 2))
+	orr	w2, w2, w0, lsl (DEBUG_UART_TX_GPIO_PORT << 2)
 	str	w2, [x1, #GPIO_AFRL_OFFSET]
 #endif
 	/* Clear UART clock flexgen divisors, keep enable bit */
diff --git a/plat/st/stm32mp2/bl31_plat_setup.c b/plat/st/stm32mp2/bl31_plat_setup.c
index b048cab96..c724313a0 100644
--- a/plat/st/stm32mp2/bl31_plat_setup.c
+++ b/plat/st/stm32mp2/bl31_plat_setup.c
@@ -185,22 +185,3 @@ entry_point_info_t *bl31_plat_get_next_image_ep_info(unsigned int type)
 
 	return NULL;
 }
-
-#if STM32MP_UART_PROGRAMMER || STM32MP_USB_PROGRAMMER
-static const plat_psci_ops_t null_psci_ops;
-
-/* Stub PSCI platform functions */
-plat_local_state_t plat_get_target_pwr_state(unsigned int lvl,
-					     const plat_local_state_t *states,
-					     unsigned int ncpu)
-{
-	return 0U;
-}
-
-int plat_setup_psci_ops(uintptr_t sec_entrypoint,
-			const plat_psci_ops_t **psci_ops)
-{
-	*psci_ops = &null_psci_ops;
-	return 0;
-}
-#endif /* STM32MP_UART_PROGRAMMER || STM32MP_USB_PROGRAMMER */
diff --git a/plat/st/stm32mp2/platform.mk b/plat/st/stm32mp2/platform.mk
index b7946024d..d2534a446 100644
--- a/plat/st/stm32mp2/platform.mk
+++ b/plat/st/stm32mp2/platform.mk
@@ -265,9 +265,7 @@ PLAT_BL_COMMON_SOURCES		+=	drivers/st/clk/clk-stm32-core.c				\
 
 PLAT_BL_COMMON_SOURCES  	+=	drivers/st/nvmem/stm32mp_tamp_nvram_mp2.c
 
-ifeq ($(filter 1,${STM32MP_UART_PROGRAMMER} ${STM32MP_USB_PROGRAMMER}),)
 PLAT_BL_COMMON_SOURCES  	+=	plat/st/stm32mp2/stm32mp2_context.c
-endif
 
 BL2_SOURCES			+=	plat/st/stm32mp2/plat_bl2_mem_params_desc.c
 
@@ -352,9 +350,7 @@ BL31_SOURCES			+=	${FDT_WRAPPERS_SOURCES}
 BL31_SOURCES			+=	plat/st/stm32mp2/bl31_plat_setup.c			\
 					plat/st/stm32mp2/stm32mp2_topology.c
 
-ifeq ($(filter 1,${STM32MP_UART_PROGRAMMER} ${STM32MP_USB_PROGRAMMER}),)
 BL31_SOURCES			+=	plat/st/stm32mp2/stm32mp2_pm.c
-endif
 
 BL31_SOURCES			+=	drivers/st/ddr/stm32mp2_ddr_helpers.c
 
